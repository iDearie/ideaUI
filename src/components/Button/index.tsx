import React, { ComponentType, Component } from 'react';
import {
  GestureResponderEvent,
  StyleProp,
  Text,
  TextStyle,
  TouchableWithoutFeedback,
  View,
  ViewStyle
} from 'react-native';
import { WithTheme } from '../Theme';
import buttonStyle from './styles';

interface ButtonProps {
  type?: 'default' | 'ghost'; // 按钮样式 默认: default 幽灵: ghost
  disabled?: boolean; // 按钮禁用状态 默认false
  disabledType?: 'neutral' | 'status'; // 按钮禁用样式 默认: neutral
  viewStyle?: StyleProp<ViewStyle>; // 按钮根节点样式
  textStyle?: StyleProp<TextStyle>; // 按钮文案样式
  bgColor?: 'base' | 'aux';
  size?: 'large' | 'default' | 'small'; // 按钮大小
  onPress?: (event: GestureResponderEvent) => void;
  children?: string;
}
const initialState = { pressIn: false };
type State = Readonly<typeof initialState>;
export class Button extends React.Component<ButtonProps, State> {
  static defaultProps: ButtonProps = {
    type: 'default',
    disabled: false,
    disabledType: 'neutral',
    bgColor: 'base',
    size: 'default'
  };

  readonly state: State = initialState;

  onPressIn = () => {
    this.setState({
      pressIn: true
    });
  };

  onPressOut = () => {
    this.setState({
      pressIn: false
    });
  };

  render() {
    const { children, type, disabled, disabledType, bgColor, size, viewStyle, textStyle, onPress } = this.props;
    const { pressIn } = this.state;
    return (
      <WithTheme themeStyles={buttonStyle}>
        {(styles) => {
          return (
            <TouchableWithoutFeedback
              onPressIn={this.onPressIn}
              onPress={onPress}
              onPressOut={this.onPressOut}
              disabled={disabled}>
              <View
                style={[
                  styles[`style_btn_${type}_container`],
                  styles[`style_btn_size_${size}`],
                  viewStyle,
                  pressIn ? styles[`style_btn_${type}_pressIn_container`] : null,
                  disabled ? styles[`style_btn_${type}_disabled_${disabledType}_container`] : null
                ]}>
                <Text
                  style={[
                    styles[`style_btn_${type}_text`],
                    styles[`style_btn_size_${size}_text`],
                    textStyle,
                    disabled ? styles[`style_btn_default_text`] : null
                  ]}>
                  {children}
                </Text>
              </View>
            </TouchableWithoutFeedback>
          );
        }}
      </WithTheme>
    );
  }
}
