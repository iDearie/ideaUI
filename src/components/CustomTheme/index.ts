export const main_bg_color = '#ff6644';
export const main_text_color = '#ff6644';
export const main_border_color = '#ff6644';
export const border_color = '#f8f8f8';
