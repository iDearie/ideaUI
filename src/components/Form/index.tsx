import * as React from 'react';
import { Text, View, ViewStyle, TextStyle, StyleProp } from 'react-native';
import { createForm } from 'rc-form';
import { styles } from './styls';
import OperaterList from '../OperaterList';
const { Item: OperaterListItem } = OperaterList;

interface FormItemProps {
    viewStyle: StyleProp<ViewStyle>; // Item 根节点样式
    labelStyle: StyleProp<TextStyle>; // Item label节点样式
    extendStyle: StyleProp<ViewStyle>; // Item 扩展组件样式
    onPress: () => void; // Item 点击方法
    separate: number; // Item 分隔符
    borderBottom?: number; // Item 下边框
    label: string; // Item label文字
    error?: string[]; // Item 错误文案
    activeOpacity?: number;
    showArror?: boolean;
    disabled?: boolean;
    touchabled?: boolean;
    children: any;
}

interface FormProps {
    style: StyleProp<ViewStyle>; // 根节点样式
}

class FormItem extends React.PureComponent<FormItemProps> {
    getError = (error: any) => {
        if (error) {
            return error.map((info: any) => {
                return (
                    <Text style={styles.errorinfoText} key={info}>
                        {info}
                    </Text>
                );
            });
        }
        return null;
    };

    renderChild = (props: FormItemProps) => {
        const { children, extendStyle, ...otherProps } = props;
        const isString = typeof children === 'string';
        return (
            <View style={[styles.style_list_item_container, !isString ? extendStyle : null]}>
                {React.Children.map(
                    !isString ? (
                        children
                    ) : (
                        <Text numberOfLines={1} style={[styles.style_extend_text, extendStyle]}>
                            {children}
                        </Text>
                    ),
                    child => {
                        return React.cloneElement(child as React.ReactElement<any>, {
                            ...otherProps
                        });
                    }
                )}
            </View>
        );
    };
    render() {
        const {
            error,
            viewStyle,
            label,
            labelStyle,
            extendStyle,
            onPress,
            separate = 0,
            borderBottom = 0.5,
            activeOpacity,
            showArror,
            disabled = false,
            touchabled = true
        } = this.props;

        return (
            <OperaterList>
                <OperaterListItem
                    touchabled={touchabled}
                    disabled={disabled}
                    borderBottom={borderBottom}
                    separate={separate}
                    onPress={onPress}
                    label={label}
                    viewStyle={viewStyle}
                    labelStyle={labelStyle}
                    extendStyle={extendStyle}
                    activeOpacity={activeOpacity}
                    showArror={showArror}>
                    {this.renderChild(this.props)}
                </OperaterListItem>
            </OperaterList>
        );
    }
}

export default class Form extends React.Component<FormProps> {
    static FormItem = FormItem;
    static create = createForm;
    render() {
        const { children, style } = this.props;
        return <View style={style}>{children}</View>;
    }
}
