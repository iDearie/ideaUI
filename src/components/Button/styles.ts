import { main_bg_color, main_border_color, main_text_color } from './../CustomTheme/index';
import { StyleSheet } from 'react-native';

const mainColor = '#ff6644';

export const styles: any = StyleSheet.create({
    style_btn_default_container: {
        backgroundColor: main_bg_color,
        paddingHorizontal: 40,
        borderRadius: 4
    },
    style_btn_size_large: {
        height: 44
    },
    style_btn_size_default: {
        height: 36
    },
    style_btn_size_small: {
        height: 32
    },
    style_btn_default_text: {
        color: '#fff',
        textAlign: 'center'
    },
    style_btn_size_large_text: {
        lineHeight: 42,
        fontSize: 16
    },
    style_btn_size_default_text: {
        lineHeight: 34,
        fontSize: 15
    },
    style_btn_size_small_text: {
        lineHeight: 30,
        fontSize: 13
    },
    style_btn_default_disabled_neutral_container: {
        backgroundColor: '#ccc'
    },
    style_btn_default_disabled_status_container: {
        backgroundColor: '#FFB2A1'
    },
    style_btn_default_prassIn_container: {
        opacity: 0.8
    },
    style_btn_ghost_container: {
        borderColor: main_border_color,
        borderWidth: 1,
        backgroundColor: '#fff',
        paddingHorizontal: 40,
        height: 42,
        borderRadius: 4
    },
    style_btn_ghost_text: {
        color: main_text_color,
        textAlign: 'center'
    },
    style_btn_ghost_disabled_neutral_container: {
        backgroundColor: '#ccc',
        borderColor: '#ccc'
    },
    style_btn_ghost_disabled_status_container: {
        backgroundColor: '#FFB2A1'
    },
    style_btn_ghost_prassIn_container: {
        backgroundColor: '#FFEFEC'
    }
});
