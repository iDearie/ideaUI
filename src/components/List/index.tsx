import * as React from 'react';
import {
  GestureResponderEvent,
  Image,
  StyleProp,
  Text,
  TextStyle,
  TouchableOpacity,
  View,
  ViewStyle
} from 'react-native';
import { styles } from './styles';

interface ListItemProps {
  viewStyle?: StyleProp<ViewStyle>; // Item组件根节点样式
  label?: string | number | React.ReactElement<any> | React.ReactElement<any>[]; // 组件左侧文字或者组件
  icon?: any; // 左侧icon
  labelStyle?: StyleProp<TextStyle> | StyleProp<ViewStyle>; // 左侧label节点样式
  extendStyle?: StyleProp<TextStyle | ViewStyle>; // 扩展组件样式
  onPress?: (event: GestureResponderEvent) => void; // 点击Item调用方法
  separate?: number; // 分隔符高度 默认为0
  borderBottom?: number; // 下边框 默认为0.5
  showArrow?: boolean; //是否显示右侧箭头
  activeOpacity?: number;
  disabled?: boolean;
  touchable?: boolean;
  children: React.ReactElement<any> | React.ReactElement<any>[] | string | number | null;
}

interface ListProps {
  style?: StyleProp<ViewStyle>;
}

class Item extends React.PureComponent<ListItemProps> {
  renderChild = (props: ListItemProps) => {
    const { children, ...otherProps } = props;
    return React.cloneElement(children as React.ReactElement, {
      ...otherProps
    });
  };

  render() {
    const {
      viewStyle,
      label,
      labelStyle,
      onPress,
      extendStyle,
      separate = 0,
      borderBottom = 0.5,
      showArrow = true,
      icon,
      activeOpacity = 0.2,
      disabled = false,
      touchable = true,
      children
    } = this.props;

    const viewContainer = (
      <View style={[styles.container, { marginBottom: separate }]}>
        <View
          style={[
            styles.style_list_item_container,
            viewStyle,
            borderBottom ? { borderBottomWidth: borderBottom } : null
          ]}>
          {label ? (
            <View style={[styles.style_label_view_container]}>
              {icon ? <Image source={icon} style={[styles.style_label_view_container_icon]} /> : null}
              {typeof label === 'string' ? (
                <Text style={[labelStyle ? labelStyle : styles.defaultLabelStyle]}>{label}</Text>
              ) : (
                <View style={[labelStyle]}>{label}</View>
              )}
            </View>
          ) : null}
          {!children || typeof children === 'string' || typeof children === 'number' ? (
            <View style={[styles.styles_extend_component_container]}>
              {children ? <Text style={[styles.styles_extend_text, extendStyle]}>{children}</Text> : null}
              {showArrow ? <Image style={[styles.arrow_png]} source={require('./arrow.png')} /> : null}
            </View>
          ) : (
            <View
              style={[
                styles.style_open_view_container,
                extendStyle,
                !label ? styles.style_open_view_container_no_label : null
              ]}>
              {this.renderChild(this.props)}
              {showArrow ? <Image style={[styles.arrow_png]} source={require('./arrow.png')} /> : null}
            </View>
          )}
        </View>
      </View>
    );

    return touchable ? (
      <TouchableOpacity disabled={disabled} onPress={onPress} activeOpacity={!onPress ? 1 : activeOpacity}>
        {viewContainer}
      </TouchableOpacity>
    ) : (
      viewContainer
    );
  }
}

export default class List extends React.Component<ListProps> {
  static Item = Item;
  render() {
    const { children, style } = this.props;
    return <View style={style}>{children}</View>;
  }
}
