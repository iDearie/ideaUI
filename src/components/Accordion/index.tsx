import React, { Component } from 'react';
import { View, Text, Animated } from 'react-native';
import { styles } from './style';
import List from '../List';
import Icon from '../Icon';

interface PanelProps {
  header: React.ReactElement | string;
  key: string;
  activeKeys?: string[];
}

class Panel extends Component<PanelProps> {
  triggerPanel = () => {};

  getChildLayout = ({ nativeEvent }: any) => {
    console.log('TCL: Panel -> getChildLayout -> layout', nativeEvent);
  };
  render() {
    const { children, header, activeKeys = [], key } = this.props;
    console.log('TCL: render -> key', this);
    return (
      <View>
        {header ? (
          <View style={[styles.header_container]}>
            <List.Item
              showArrow={false}
              borderBottom={0}
              onPress={this.triggerPanel}
              label={React.isValidElement(header) ? header : <Text>{header}</Text>}>
              <Icon name={'chevron-down'} />
            </List.Item>
          </View>
        ) : null}
        <View>{children}</View>
      </View>
    );
  }
}

interface AccordionProps {
  activeKeys?: string[];
}

export default class Accordion extends Component<AccordionProps> {
  static Panel = Panel;
  render() {
    const { children, activeKeys } = this.props;
    return React.Children.map(children, (child) => {
      return React.cloneElement(child as React.ReactElement, { activeKeys });
    });
  }
}
