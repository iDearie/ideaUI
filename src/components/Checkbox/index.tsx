import * as React from 'react';
import {
  Text,
  TouchableOpacity,
  View,
  TextStyle,
  ViewStyle,
  Image,
  TouchableWithoutFeedback,
  StyleProp
} from 'react-native';
import { styles } from './style';
import List from '../List';

export interface CheckboxProps {
  style?: StyleProp<ViewStyle>; // 根节点style
  type?: 'checkbox' | 'tag'; // checkbox 类型   checkbox: 正常  tag: 带边框
  layout?: 'list' | 'inline'; // 布局 list: 竖向布局   inline: 行内布局
  onChange?: (activeKey: number, checkedValue: number[]) => void; // 改变状态回调
  radio?: boolean; // 是否单选 默认false
  defaultValue?: number[] | string[]; // 默认值
  value: string; // Item 组件 value
  children: any;
}

export interface CheckboxItemProps {
  textStyle?: StyleProp<TextStyle>; // Item: 文本节点样式
  viewStyle?: StyleProp<ViewStyle>; // Item 组件 根节点样式
  value: string; // Item 组件 value
  type?: 'checkbox' | 'tag';
  layout?: 'list' | 'inline';
  borderBottom?: number; // list 布局下的下边框
  onChange?: (activeKey: any) => void;
  checkedValues?: Array<string>; // 选中的value集合
  numberOfLines?: number;
}

class Item extends React.Component<CheckboxItemProps> {
  onPress = () => {
    const { onChange, value } = this.props;
    onChange && onChange(value);
  };
  render() {
    const {
      textStyle,
      viewStyle,
      children,
      type = 'tag',
      checkedValues = [],
      value,
      layout = 'inline',
      borderBottom = 0.5,
      numberOfLines = 1
    } = this.props;
    const checked = checkedValues.includes(value);

    const ItemStyle: any = {
      inline_tag: (
        <View
          style={[
            styles.style_checkbox_inline_tag_item_view,
            checked ? styles.style_checkbox_inline_tag_item_view_active : null,
            viewStyle
          ]}>
          <Text
            style={[
              styles.style_checkbox_inline_tag_item_text,
              checked ? styles.style_checkbox_inline_tag_item_text_active : null,
              textStyle
            ]}
            numberOfLines={numberOfLines}>
            {children}
          </Text>
        </View>
      ),
      inline_checkbox: (
        <View style={[styles.style_checkbox_inline_checkbox_item_view, viewStyle]}>
          <Image
            style={[styles.style_checkbox_inline_checkbox_item_image]}
            source={checked ? require('./image/gender_clicked.png') : require('./image/gender_normal.png')}
          />

          <Text numberOfLines={numberOfLines} style={[styles.style_checkbox_inline_checkbox_item_text, textStyle]}>
            {children}
          </Text>
        </View>
      ),
      list_checkbox: (
        <List.Item
          borderBottom={borderBottom}
          label={<Text numberOfLines={numberOfLines}>{children}</Text>}
          showArrow={false}
          onPress={this.onPress}>
          {checked ? (
            <Image style={{ width: 14, height: 14 }} source={require('./image/item_check_auto_say_hi.png')} />
          ) : null}
        </List.Item>
      )
    };

    return (
      <TouchableWithoutFeedback key={value} onPress={this.onPress}>
        {ItemStyle[`${layout}_${type}`] || null}
      </TouchableWithoutFeedback>
    );
  }
}

export default class Checkbox extends React.Component<CheckboxProps, any> {
  static Item = Item;

  constructor(props: CheckboxProps) {
    super(props);
    const { defaultValue, value } = props;
    this.state = {
      checkedValues: value || defaultValue || []
    };
  }

  componentWillReceiveProps(nextProps: CheckboxProps) {
    const { defaultValue, value } = nextProps;
    this.setState({
      checkedValues: value || defaultValue || []
    });
  }

  onChange = (activeKey: number) => {
    const { onChange, radio } = this.props;
    const { checkedValues } = this.state;
    const _checkedValues = !radio
      ? checkedValues.includes(activeKey)
        ? checkedValues.filter((v: any) => v !== activeKey)
        : [...checkedValues, activeKey]
      : [activeKey];
    onChange && onChange(activeKey, _checkedValues);
    this.setState({
      checkedValues: _checkedValues
    });
  };

  renderChild = (props: CheckboxProps) => {
    const { children, type, layout } = props;
    const { checkedValues } = this.state;
    return React.Children.map(children, (child) => {
      return React.cloneElement(child as React.ReactElement<any>, {
        type,
        layout,
        onChange: this.onChange,
        checkedValues
      });
    });
  };

  warpStyle = ({ style }: { style: StyleProp<ViewStyle> }) => {
    return {
      list: <View style={[styles[`style_checkbox_list`], style]}>{this.renderChild(this.props)}</View>,
      inline: <View style={[styles[`style_checkbox_inline`], style]}>{this.renderChild(this.props)}</View>
    };
  };

  render() {
    const { style, layout = 'list' } = this.props;

    return this.warpStyle({ style })[layout];
  }
}
