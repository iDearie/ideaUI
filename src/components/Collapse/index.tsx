import React, { Component, SyntheticEvent } from 'react';
import { Text, View } from 'react-native';
import Icon from '../Icon';
import List from '../List';
import { WithTheme } from '../Theme';
import { styles } from './style';

interface PanelProps extends CollapseProps {
  header: React.ReactElement | string;
  value: string;
}

class Panel extends Component<PanelProps> {
  height: number;

  state: {
    opened: boolean;
  };

  constructor(props: PanelProps) {
    super(props);
    this.height = 0;
    const { activeKeys = [], value } = props;
    this.state = {
      opened: activeKeys.includes(value)
    };
  }

  triggerPanel = () => {
    const { onChange, activeKeys = [], value } = this.props;
    const { opened } = this.state;
    this.setState({
      opened: !opened
    });
    onChange && onChange(opened ? activeKeys.filter((v) => v !== value) : [...activeKeys, value]);
  };

  componentWillReceiveProps(nextProps: PanelProps) {
    const { activeKeys = [], value } = nextProps;
    this.setState({
      opened: activeKeys.includes(value)
    });
  }

  render() {
    const { children, header } = this.props;
    const { opened } = this.state;
    return (
      <WithTheme themeStyles={styles}>
        {(_style) => (
          <View>
            {header ? (
              <View style={[_style.header_container]}>
                <List.Item
                  showArrow={false}
                  borderBottom={0}
                  separate={1}
                  activeOpacity={1}
                  onPress={this.triggerPanel}
                  label={React.isValidElement(header) ? header : <Text>{header}</Text>}>
                  <Icon
                    name={'chevron-down'}
                    style={[_style.icon, { transform: [{ rotate: opened ? '180deg' : '0deg' }] }]}
                  />
                </List.Item>
              </View>
            ) : null}
            {opened ? <View style={[{ height: 'auto' }, _style.panel_child_view]}>{children}</View> : null}
          </View>
        )}
      </WithTheme>
    );
  }
}

interface CollapseProps {
  activeKeys?: string[];
  onChange?: (activeKeys: string[]) => void;
}

export default class Collapse extends Component<CollapseProps> {
  static Panel = Panel;
  render() {
    const { children, activeKeys } = this.props;
    return React.Children.map(children, (child) => {
      return React.cloneElement(child as React.ReactElement, { activeKeys, onChange: this.props.onChange });
    });
  }
}
