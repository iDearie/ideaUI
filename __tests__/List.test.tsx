import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { TouchableOpacity, Image } from 'react-native';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import { Button } from '../src/components/Button';
import List from '../src/components/List';

describe('List', () => {
  it('组件快照', () => {
    const renderedValue = renderer
      .create(
        <List>
          <List.Item label={'label'}>
            <Button />
          </List.Item>
        </List>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  describe('rendering', () => {
    let wrapper: ReactWrapper;
    beforeEach(() => {
      wrapper = mount(
        <List.Item label={'Label'}>
          <Button />
        </List.Item>
      );
    });

    it('TouchableOpacity 标签渲染正确', () => {
      expect(wrapper.find(TouchableOpacity)).toHaveLength(1);
      const onPress = sinon.spy();
      wrapper.setProps({ onPress });
      const props: any = wrapper.props();
      props.onPress();
      expect(onPress.calledOnce).toBe(true);
      wrapper.setProps({ touchable: false });
      expect(wrapper.find(TouchableOpacity)).toHaveLength(0);
      props.onPress();
      expect(onPress.calledOnce).toBe(false);
    });

    it('children渲染正确', () => {
      expect(wrapper.find(Button)).toHaveLength(1);
    });

    it('label 渲染正确', () => {
      expect(
        wrapper.findWhere((node) => {
          const text = node.text();
          const type = node.type();
          return type === 'Text' && text === 'Label';
        })
      ).toHaveLength(1);
    });

    it('箭头渲染正确', () => {
      const mapItem = (node: ReactWrapper<any, any>) => {
        const type = node.type();
        const source = node.prop('source') && node.prop('source').testUri;
        return type === 'Image' && source === '../../../src/components/List/arrow.png';
      };
      expect(wrapper.findWhere(mapItem)).toHaveLength(1);
      wrapper.setProps({ showArrow: false });
      expect(wrapper.findWhere(mapItem)).toHaveLength(0);
    });

    it('下边框渲染正确', () => {
      wrapper.setProps({ borderBottom: 1 });
      const borderBottomWrapper = wrapper.findWhere((node) => {
        const type = node.type();
        const style = node.prop('style');
        return type === 'View' && style[2] && style[2].borderBottomWidth === 1;
      });
      expect(borderBottomWrapper).toHaveLength(1);
    });
  });
});
