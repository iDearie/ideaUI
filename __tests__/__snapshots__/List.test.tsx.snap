// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List 组件快照 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "marginBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomColor": "#eaeaea",
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "marginHorizontal": 16,
            },
            undefined,
            Object {
              "borderBottomWidth": 0.5,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginRight": 10,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333",
                  "fontSize": 14,
                },
              ]
            }
          >
            label
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
              undefined,
              null,
            ]
          }
        >
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ff6644",
                  "borderRadius": 4,
                },
                Object {
                  "height": 36,
                },
                undefined,
                null,
                null,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 15,
                    "lineHeight": 34,
                  },
                  undefined,
                  null,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
